apps/namadillo/src/hooks/useWasmHashes.tsx

```diff
@@ -0,0 +1,56 @@
+import { Namada, useIntegration } from "@namada/integrations";
+import { WasmHash } from "@namada/types";
+import { chainAtom, nativeTokenAddressAtom } from "atoms/chain";
+import { rpcUrlAtom } from "atoms/settings";
+import { useAtomValue } from "jotai";
+import { useCallback, useEffect, useState } from "react";
+import { getSdkInstance } from "./useSdk";
+
+export const useWasmHashes = (): void => {
```

Comment:
I think this whole hook could be refactored into an atom. chain, rpcUrl, and  nativeToken could be added as a list of dependencies for this new atom. It would be good to add this in the AppSetup component, so we can show a loading or handle the error right in the initialization

Reply:
Yeah, I was starting to think the same thing. Will take a look!